Add quantized conv3d function

xiaomengy opened the pull request:
Summary: Add quantized conv3d function

Test Plan: buck test mode/dev-nosan //caffe2/test:quauntized -- "conv"

Differential Revision: D18463090

BIT-silence commented:
link to https://github.com/pytorch/pytorch/issues/28480

jianyuh commented on torch/nn/quantized/functional.py:
```diff
@@ -118,6 +118,63 @@ def conv2d(input, weight, bias,
+        input: quantized input tensor of shape
+          :math:`(\text{minibatch} , \text{in\_channels} , iD , iH , iW)`
+        weight: quantized filters of shape
+          :math:`(\text{out\_channels} , \frac{\text{in\_channels}}{\text{groups}} , kD , kH , kW)`
```
Want to check: is it N, C, H, W, D, or N, C, D, H, W?

jianyuh replied:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qconv.cpp#L284 Here it is H, W, D. But I also see https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/cpu/qconv.cpp#L124 Here is D, H, W.

BIT-silence replied:
That is NCDHW, the first two dims are N and C, so D is at dim(2).

jianyuh commented on test/test_quantized_nn_mods.py:
```diff
@@ -49,53 +166,80 @@ def test_conv_api(self, use_bias, per_channel, qengine):
+        X_zero_point, W_scale, W_zero_point, Y_scale, Y_zero_point, use_bias,
+        use_channelwise, qengine,
+    ):
+        """Tests the correctness of the conv3d function.
```
Maybe also add a comment here conv2d is only supported by FBGEMM, not QNNPACK backend?

jianyuh reviewed (approved):
Looks good. Please land after it passes all OSS tests.

xiaomengy committed:
Add quantized conv3d function (#29686)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/29686

Add quantized conv3d function

Test Plan: buck test mode/dev-nosan //caffe2/test:quauntized -- "conv"

Reviewed By: hl475

Differential Revision: D18463090

fbshipit-source-id: 7c931712eff531fbf2ea36e5593e7b4995b0fe4a
